xml/tuning_cgroups.xml

```diff
@@ -221,8 +220,130 @@
+  <para>
+       This is the same as having no limit. It is not a requirement to change the 
+       default, but setting some limits may help to prevent system crashes
+       from runaway processes.
+   </para>
+   <para>
+       Change the global default in
+       <filename>/etc/systemd/system.conf</filename> by uncommenting the 
+       <literal>DefaultTasksMax</literal> line, and changing the value either
+       to a number of tasks, or a percentage, like the following examples:
+   </para>
+   <screen>DefaultTasksMax=128
+DefaultTasksMax=7%
+</screen>
+   <para>
+       Load the new setting, then verify that it changed:
+   </para>
+   <screen>&prompt.sudo;systemctl daemon-reload
+&prompt.user;systemctl show --property DefaultTasksMax
+DefaultTasksMax=128
+   </screen>
+   <para>
+       Now you can set higher limits on individual services as needed. This
+       example is for MariaDB. <command>systemctl</command> has at least two
+       ways of checking the current value:
+   </para>
+   <screen>&prompt.user;systemctl show -p TasksMax mariadb.service
+TasksMax=128
+
+&prompt.user;systemctl status mariadb.service
+  ● mariadb.service - MariaDB database server
+   Loaded: loaded (/usr/lib/systemd/system/mariadb.service; disabled; vendor pr>
+   Active: active (running) since Tue 2020-05-26 14:15:03 PDT; 27min ago
+     Docs: man:mysqld(8)
+           https://mariadb.com/kb/en/library/systemd/
+ Main PID: 11845 (mysqld)
+   Status: "Taking your SQL requests now..."
+    Tasks: 30 (limit: 128)
+   CGroup: /system.slice/mariadb.service
+           └─11845 /usr/sbin/mysqld --defaults-file=/etc/my.cnf --user=mysql
+   </screen>
+   <para>
+       The Tasks line shows that MariaDB currently has 30 tasks running, and
+       an upper limit of 128. The following example demonstrates how to raise
+       MariaDB's limit to 1024:
+   </para>
+   <screen>
+&prompt.sudo;systemctl edit --system mariadb.service
+[System]
+TasksMax=1024
+</screen>
+   <para>
+       This creates an override file, 
+       <filename>/etc/systemd/system/mariadb.service.d/override.conf</filename>.
+       If you prefer to edit the full service file, use the 
+       <command>--full</command> option:
+   </para>
+   <screen>&prompt.sudo;systemctl edit --system --full mariadb.service</screen>
+   <para>    
+       Then load the new value:
+   </para>
+   <screen>&prompt.sudo;systemctl daemon-reload</screen>
+   <para>
+       You may also wish to set custom values for user processes. The &sle; default
+       for user slices is also infinity, set in
+       <filename>/usr/lib/systemd/system/user-.slice.d/20-suse-defaults.conf</filename>.
+       Verify this the same way as for system services:
+   </para>
+   <screen>&prompt.user;systemctl show -p TasksMax user-1000.slice
+TasksMax=18446744073709551615
+   </screen>
+   <para>
+       Change this for <literal>user-1000.slice</literal> to 256:
+   </para>
+   <screen>&prompt.sudo;systemctl edit user@1001.service
+[Service]
```

Comment:
I think the example you gave is a bit misleading because it only set a limits on the number of processes spawned by the systemd user instance.

In order to limit the number of *all* processes spawned by a user (which includes all processes started from user sessions) you need to set the limit on the user's slice instead. For example:

`cat /etc/systemd/system/user-1000.slice.d/50-tasksmax.conf
[Slice]
TasksMax=256`

Reply:
Also please note that this only works with SLE15 or later.